File: kanbanery.py
import urllib
import urllib2
import base64

class Kanbanery:
  def __init__(self, workspace, project):
    self.workspace = workspace
    self.project = project
    self.key = None
    self.email = None
    self.password = None

  def create_bug(self, title, body):
    url = "https://{workspace}.kanbanery.com/api/v1/projects/{project}/tasks.json".format(workspace=self.workspace, project=self.project)

    data = urllib.urlencode({ "task[task_type_name]" : "bug",
                              "task[title]" : title,
                              "task[description]" : body})

    header = {}
    if self.key:
      header = { "X-Kanbanery-ApiToken" : self.key }
    else:
      base64string = base64.encodestring('%s:%s' % (self.email, self.password)).replace('\n', '')
      header = { "Authorization" : "Basic %s" % base64string }
    
    try: 
      request = urllib2.Request(url, data, header)
      response = urllib2.urlopen(request)
      return [str(response.code) + ' OK', response.read()]
    except urllib2.URLError, e:
      return [str(e.code) + ' ERROR', e.read()]

File: kanbanery.rb
require 'net/http'
require 'net/https'

class Kanbanery
  attr_accessor :key, :email, :password, :workspace, :project

  def initialize(workspace, project)
    @workspace = workspace
    @project = project
  end

  def create_bug(title, body)
    http = Net::HTTP.new("#{@workspace}.kanbanery.com", 443)
    http.use_ssl = true

    req = Net::HTTP::Post.new("/api/v1/projects/#{@project}/tasks.json")
    req.set_form_data({"task[task_type_name]" => "bug",
                       "task[title]" => title,
                       "task[description]" => body}, ";")

    if @key
      req["X-Kanbanery-ApiToken"] = @key
    else
      req.basic_auth @email, @password
    end

    res = http.request(req)
    [res.code, res.body]
  end
end
